include/s3select.h

```diff
@@ -2238,10 +2251,58 @@ class base_s3object
       m_where_clause->traverse_and_apply(m_sa, m_s3_select->get_aliases(), m_s3_select->is_json_query());
     }
 
+    for (auto& a : *m_s3_select->get_aliases()->get())
+    {
+        m_alias_keys.push_back(a.first);
+        set.insert(a.second);
+        col_ref = true;
+        is_alias = true;
+    }
```

Comment:
`a.second` is a pointer to projection.
thus
upon inserted into `set`, it contains a unique element 
`set.count`(a.second) will return 1. ( or 0 upon other pointers)
thus, why there is a need for `col_ref` or `is_alias` ?
`set` contains all the information.


set is not a good name, please use some other name that describes the content of set.